tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Kinesis/KinesisContextAdapter.cs

```diff
@@ -0,0 +1,87 @@
+// <copyright file="KinesisContextAdapter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Datadog.Trace.Headers;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.Kinesis
+{
+    internal struct KinesisContextAdapter : IHeadersCollection, IBinaryHeadersCollection
+    {
+        private static readonly IDatadogLogger Logger = DatadogLogging.GetLoggerFor<KinesisContextAdapter>();
+        private Dictionary<string, List<string>> headers = new Dictionary<string, List<string>>();
+
+        public KinesisContextAdapter()
+        {
+        }
+
+        public Dictionary<string, object> GetDictionary()
+        {
+            var retval = new Dictionary<string, object>();
+            foreach (string key in headers.Keys)
+            {
+                retval.Add(key, headers[key][0]);
+            }
+
+            return retval;
```

Comment:
another thing I'm thinking is that this can probably be written in a clearer way with Linq as
```suggestion
            return headers.ToDictionary(kvp => kvp.Key, kvp => (object)kvp.Value);
```